src/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoom.java

```diff
@@ -373,7 +373,7 @@ public LocalMUCRoom() {
         this.canChangeNickname = MUCPersistenceManager.getBooleanProperty(mucService.getServiceName(), "room.canChangeNickname", true);
         this.registrationEnabled = MUCPersistenceManager.getBooleanProperty(mucService.getServiceName(), "room.registrationEnabled", true);
         // TODO Allow to set the history strategy from the configuration form?
-        roomHistory = new MUCRoomHistory(this, new HistoryStrategy(mucService.getHistoryStrategy()));
+        roomHistory = new MUCRoomHistory(this, mucService.getHistoryStrategy());

```

Comment:
Yes, I think you are absolutely right. I've overlooked that. `HistoryStrategy` stores the messages as well. I think to solve it, `roomSubject` and the messages should be attached to `LocalMUCRoom` maybe, and `HistoryStrategy` should only hold information about the actual strategy.